### ISSUE TITLE
Flagship cannot plan wormhole travel without hyper/jump drive using mouse clicks

### ISSUE THREAD

**Issue by tehhowch** [endless-sky/endless-sky#2936]
Per the title. If the player's flagship has no Hyperdrive or Jump Drive, the mouse cannot be used to plan travel over otherwise valid wormhole routes.
Ref: https://steamcommunity.com/app/404410/discussions/1/1471967615874008822/#c1471967615881475355

Detail: 
MapPanel initializes DistanceMap with the [current player](https://github.com/endless-sky/endless-sky/blob/v0.9.8/source/MapPanel.cpp#L58), but due to the [lack of a hyper/jump drive, the initialization of routes is aborted](https://github.com/endless-sky/endless-sky/blob/v0.9.8/source/DistanceMap.cpp#L157), and the "You have not yet mapped a route to this system" [message is displayed](https://github.com/endless-sky/endless-sky/blob/v0.9.8/source/MapPanel.cpp#L102) on top of the map, even if the player has visited the system / used the wormhole.
The system can still be [selected via the J key](https://github.com/endless-sky/endless-sky/blob/v0.9.8/source/MapDetailPanel.cpp#L162), which shortcuts [the check in Select](https://github.com/endless-sky/endless-sky/blob/v0.9.8/source/MapPanel.cpp#L446) (but of course the "You have not yet mapped a route to this system" message is still displayed).

This is currently only a mod issue, as there are no systems in the base game with an Outfitter and also a wormhole.

**Comment by tehhowch** [comment]
I think the simplest fix for this is changing DistanceMap::Init line 157 from
```
if(!hyperspaceFuel && !jumpFuel)
```
to
```
if(!hyperspaceFuel && !jumpFuel && !player)
```
This shouldn't affect any of the AI escort pathfinding while allowing the player to design wormhole-only travel plans via both click / J key and not receive the "You have not..." message.

**Comment by tehhowch** [comment]
This is fixed by #2938, which checks the local system for a wormhole when initializing DistanceMap, and only exits early if there is no wormhole and no jump method.
(will close with milestone)
